dot/rpc/modules/payment_test.go

```diff
@@ -14,41 +14,44 @@ import (
 	"github.com/ChainSafe/gossamer/lib/common"
 	mocksruntime "github.com/ChainSafe/gossamer/lib/runtime/mocks"
 	"github.com/ChainSafe/gossamer/pkg/scale"
+	"github.com/golang/mock/gomock"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 func TestPaymentModule_QueryInfo(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
 	testHash := common.NewHash([]byte{0x01, 0x02})
 	u, err := scale.NewUint128(new(big.Int).SetBytes([]byte{1, 2, 3, 4, 5, 6, 7, 8, 1, 2, 3, 4, 5, 6}))
 	require.NoError(t, err)
 
-	runtimeMock := mocksruntime.NewInstance(t)
-	runtimeMock2 := mocksruntime.NewInstance(t)
-	runtimeErrorMock := mocksruntime.NewInstance(t)
+	runtimeMock := mocksruntime.NewMockInstance(ctrl)
+	runtimeMock2 := mocksruntime.NewMockInstance(ctrl)
+	runtimeErrorMock := mocksruntime.NewMockInstance(ctrl)
 
-	blockAPIMock := mocks.NewBlockAPI(t)
-	blockAPIMock2 := mocks.NewBlockAPI(t)
-	blockErrorAPIMock1 := mocks.NewBlockAPI(t)
-	blockErrorAPIMock2 := mocks.NewBlockAPI(t)
+	blockAPIMock := mocks.NewMockBlockAPI(ctrl)
+	blockAPIMock2 := mocks.NewMockBlockAPI(ctrl)
+	blockErrorAPIMock1 := mocks.NewMockBlockAPI(ctrl)
+	blockErrorAPIMock2 := mocks.NewMockBlockAPI(ctrl)
 
-	blockAPIMock.On("BestBlockHash").Return(testHash, nil)
-	blockAPIMock.On("GetRuntime", testHash).Return(runtimeMock, nil)
+	blockAPIMock.EXPECT().BestBlockHash().Return(testHash).AnyTimes()
+	blockAPIMock.EXPECT().GetRuntime(testHash).Return(runtimeMock, nil).AnyTimes()
```

Comment:
Are we just using AnyTimes() because its already such a big PR?

Reply:
Yep and because previous mockery mocks weren't checking the mock was called a precise number of times.
I created https://github.com/ChainSafe/gossamer/issues/3038 to address this at some point